File: src/main/java/org/eln/sim/process/destruct/ResistorPowerWatchdog.java
package org.eln.sim.process.destruct;

/*
NOTE: DO NOT IMPORT MINECRAFT CODE IN THIS CLASS
EXTEND IT INSTEAD IN THE org.eln.nbt DIRECTORY
 */

import org.eln.sim.mna.component.Resistor;

public class ResistorPowerWatchdog extends ValueWatchdog {

    Resistor resistor;

    public ResistorPowerWatchdog set(Resistor resistor) {
        this.resistor = resistor;
        return this;
    }

    public ResistorPowerWatchdog setPmax(double Pmax) {
        this.max = Pmax;
        this.min = -1;
        this.timeoutReset = Pmax * 0.20 * 5;

        return this;
    }

    @Override
    double getValue() {
        return resistor.getP();
    }
}

File: src/main/java/org/eln2/sim/process/destruct/ResistorPowerWatchdog.kt
package org.eln2.sim.process.destruct

/*
NOTE: DO NOT IMPORT MINECRAFT CODE IN THIS CLASS
EXTEND IT INSTEAD IN THE org.eln.nbt DIRECTORY
 */

import org.eln2.sim.mna.component.Resistor

class ResistorPowerWatchdog(var resistor: Resistor): ValueWatchdog() {
    var maximumPower: Double
        get(): Double {
            return max
        }
        set(mp) {
            max = mp * 1.2
            min = -1.0
        }

    override fun getValue(): Double {
        return resistor.getP()
    }
}